src/base/components/DeviceProfile.js

```diff
@@ -43,10 +102,16 @@ class DeviveProfileTab extends React.Component {
         return resp.json()
       })
       .then((data) => {
-        if(this.state.devicestatus === "on")
-          this.setState({ devicestatus: "off"});
-        else
-        this.setState({ devicestatus: "on"});
+        if(data.status){
+          if(this.state.devicestatus === "on" )
+            this.setState({ devicestatus: "off"});
+          else
+          this.setState({ devicestatus: "on"});
+        }
+        else{
+          this.setState({ devicestatus: "on"});
+          toast.error("Device seems offline, Please start Experiment!!!");
```

Comment:
The device is offline. Please re-start the experiment.

Reply:
Done